fix(solutions-telco): BE->AE spelling, new form to get in touch [WD-17656]

lorumic opened the pull request:
## Done

- Updated the [/solutions/telco](https://canonical-com-1485.demos.haus/solutions/telco) page according to the [copy doc](https://docs.google.com/document/d/1yRSYYY8_UNfZzdLcp_fYxi-STGm40o3UP9vD2vW6brw/edit?tab=t.0).
- Added new contact form according to the [copy doc](https://docs.google.com/document/d/1Q-w36nbiNrNuTXh5xl9zrkdb9Br5KGiROkgsYlOX7GQ/edit?tab=t.0), instead of redirecting to a form on ubuntu.com.

## QA

- Open https://canonical-com-1485.demos.haus/solutions/telco
- Check that all the recent suggestions in the copy doc linked above have been addressed
- Check that the new contact form works, and does not redirect to ubuntu.com

## Issue / Card

Fixes [WD-17656](https://warthogs.atlassian.net/browse/WD-17656)

[WD-17656]: https://warthogs.atlassian.net/browse/WD-17656?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://canonical-com-1485.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/9072)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

britneywwc commented on templates/solutions/telco/index.html:
```diff
@@ -618,13 +618,14 @@ <h3 class="p-heading--5">Contact us</h3>
   </section>
 
   <script src="{{ versioned_static('js/tabbed-content.js') }}"></script>
+  <script defer src="{{ versioned_static('js/modals.js') }}"></script>
```
Form template also has to be imported in the respective files
```suggestion
  <script defer src="{{ versioned_static('js/modals.js') }}"></script>
  {% include "/shared/forms/form-template.html" %}
```

lorumic replied:
Thanks for the suggestion - that solved the issue indeed. :)

britneywwc reviewed (commented):
Thanks for using the form templates. There are two things that you need for the modals to work:
1) Import `{% include "/shared/forms/form-template.html" %}` on all the static files that are using the form templates, this includes the static children pages such as /solutions/telco/5g-core.html
2) On the children pages, you have to add `aria-controls="solutions-telco-modal"` to CTA buttons and links to invoke the modal.

I realised that these were missing on the docs, so I will update them accordingly. Let me know again if this doesn't work

eliman11 commented:
Thanks @lorumic! Just realized there's some words that are still spelled in British English, I've flagged them here and on the copydoc: 
- [ ] Change "centralised" to "centralized" in "Canonical offers secure, modular and interoperable solutions. Ideal for public and private clouds, multi-cloud or hybrid-cloud environments at centralized or distributed sites."
- [ ] Optimise -> Optimize in "Ubuntu is also widely tested on the world’s leading silicon to optimize resource consumption."
- [ ] Centres -> Centers in "Choose Canonical OpenStack for data centers or Microcloud for small private clouds and edge environments."

Also,
- [ ] Instead of the "Tell us more about.." field, I think the PM wanted a "What would you like to talk with us about?" text field after the job input field - I'm referencing [this copydoc](https://docs.google.com/document/d/1Q-w36nbiNrNuTXh5xl9zrkdb9Br5KGiROkgsYlOX7GQ/edit?tab=t.0) for the form 
<img width="1348" alt="Screenshot 2025-01-09 at 11 10 08" src="https://github.com/user-attachments/assets/50a5721e-c2ac-4377-a661-52d3cf7fb549" />

- [ ] Country shouldn't be a mandatory field per the copydoc

lorumic commented:
> * [ ]  Country shouldn't be a mandatory field per the copydoc

@britneywwc Can you help with this one? Currently, country stays a required field even if I set `isRequired` to `false` in the the `forms-data.json` file. Is this intentional (i.e. country should always be mandatory) or is it a bug?

@eliman11 Thanks for the quick review! I have addressed your other points. 2 of the AE updates that you mentioned above were missing in the copy doc, but I had already addressed them before. Would you mind having another look?

britneywwc commented:
@lorumic I just pushed a fix that applies the country's `isRequired` field as expected

lorumic commented:
> @lorumic I just pushed a fix that applies the country's `isRequired` field as expected

Thanks for the quick fix!
@eliman11 All your points should have been addressed now, can you please have another look?

eliman11 commented:
Thanks for the changes! My bad - missed one more US spelling update -> "[Deploy your data centre with MAAS ›](https://maas.io/)" should be "center". Would we be able to move the message field below the contact details? So it would look something like this contact form: 
![image](https://github.com/user-attachments/assets/89932e2b-b4f4-4aec-b62b-2d09cfedf57b)

lorumic commented:
> Thanks for the changes! My bad - missed one more US spelling update -> "[Deploy your data centre with MAAS ›](https://maas.io/)" should be "center". Would we be able to move the message field below the contact details? So it would look something like this contact form:

Sure thing, I've updated that spelling and made the form look like the one you shared.

eliman11 commented:
Looks great, thanks for the quick turnaround! Adding UX+1

lorumic commented:
> The spacing below 'phone number' seems a little off.

Hey Pete! Thanks for the review. You're right, the spacing was off - the "culprit" was the intl-tel-input library, which forced a margin-bottom of 0 on the phone input. So I moved the standard Vanilla spacing for input elements to the wrapper of that input element.

The advantage of this approach is that it will fix the issue everywhere else - for example, in [the contact form on the home page](https://canonical.com/#get-in-touch), which also shows the same spacing issue at the moment. Also, it should keep any other stakeholder (UX, PdM) equally happy, because the order they set for the items in the form won't be altered. :)

Let me know what you think about it, thanks!

petesfrench commented:
@lorumic Great find! This is goes above and beyond what we expect from the people outside the team when dealing with BAU. So needless to say I am very grateful for this extra work!

Let me know when you have addressed my other comments and we can get this PR merged

lorumic commented:
> Let me know when you have addressed my other comments and we can get this PR merged

Which ones? I see only that one:

![image](https://github.com/user-attachments/assets/69b246d4-3d94-4c37-9570-50bb3d03275e)

petesfrench commented:
@lorumic Apologies, I have not confirmed them

lorumic commented on templates/shared/forms/form-fields.html:
```diff
@@ -20,7 +20,12 @@
                 {% for field in fieldset.fields %}
                   {% if field.type == "country" %}
-                    {% include "shared/forms/_country.html" %}
+                    {% with %}
+                      {% if not field.isRequired %}
+                        {% set required = false %}
+                        {% endif %}
+                      {% include "shared/forms/_country.html" %}
+                    {% endwith %}
```
I'm no expert of jinja, would the following fix the misalignment?
```suggestion
                    {% with %}
                      {% if not field.isRequired %}
                        {% set required = false %}
                      {% endif %}
                      {% include "shared/forms/_country.html" %}
                    {% endwith %}
```

lorumic commented on static/files/forms-data.json:
```diff
@@ -334,6 +334,77 @@
+              "type": "country",
+              "id": "country",
+              "label": "Country",
+              "isRequired": false
```
True, removed

lorumic commented:
> @lorumic Apologies, I have not confirmed them

No problem - addressed both :)

petesfrench reviewed (approved):
LGTM, thanks for this work @lorumic !
